ros/src/computing/planning/mission/packages/op_global_planner/nodes/op_global_planner_core.cpp

```diff
@@ -0,0 +1,392 @@
+		marker.pose.orientation = tf::createQuaternionMsgFromYaw(destinations.at(i).pos.a);
+		goals_array.markers.push_back(marker);
+	}
+	pub_GoalsListRviz.publish(goals_array);
+}
+
+void GlobalPlanner::SaveSimulationData()
+{
+	std::vector<std::string> simulationDataPoints;
+	std::ostringstream startStr;
+	startStr << m_CurrentPose.pos.x << "," << m_CurrentPose.pos.y << "," << m_CurrentPose.pos.z << "," << m_CurrentPose.pos.a << ","<< m_CurrentPose.cost << "," << 0 << ",";
+	simulationDataPoints.push_back(startStr.str());
+
+	for(unsigned int i=0; i < m_GoalsPos.size(); i++)
+	{
+		std::ostringstream goalStr;
+		goalStr << m_GoalsPos.at(i).pos.x << "," << m_GoalsPos.at(i).pos.y << "," << m_GoalsPos.at(i).pos.z << "," << m_GoalsPos.at(i).pos.a << "," << 0 << "," << 0 << ",destination_" << i+1 << ",";
+		simulationDataPoints.push_back(goalStr.str());
+	}
+
+	std::string header = "X,Y,Z,A,C,V,name,";
+
+	std::ostringstream fileName;
+	fileName << UtilityHNS::UtilityH::GetHomeDirectory()+UtilityHNS::DataRW::LoggingMainfolderName+UtilityHNS::DataRW::SimulationFolderName;
+	fileName << "EgoCar.csv";
+	std::ofstream f(fileName.str().c_str());
+
+	if(f.is_open())
+	{
+		if(header.size() > 0)
+			f << header << "\r\n";
+		for(unsigned int i = 0 ; i < simulationDataPoints.size(); i++)
+			f << simulationDataPoints.at(i) << "\r\n";
+	}
+
+	f.close();
+}
+
+int GlobalPlanner::LoadSimulationData()
+{
+	std::ostringstream fileName;
+	fileName << "EgoCar.csv";
+
+	std::string simuDataFileName = UtilityHNS::UtilityH::GetHomeDirectory()+UtilityHNS::DataRW::LoggingMainfolderName+UtilityHNS::DataRW::SimulationFolderName + fileName.str();
+	UtilityHNS::SimulationFileReader sfr(simuDataFileName);
+	UtilityHNS::SimulationFileReader::SimulationData data;
+
+	int nData = sfr.ReadAllData(data);
+	if(nData == 0)
+		return 0;
+
+	m_CurrentPose = PlannerHNS::WayPoint(data.startPoint.x, data.startPoint.y, data.startPoint.z, data.startPoint.a);
+	m_GoalsPos.push_back(PlannerHNS::WayPoint(data.goalPoint.x, data.goalPoint.y, data.goalPoint.z, data.goalPoint.a));
+
+	for(unsigned int i=0; i < data.simuCars.size(); i++)
+	{
+		m_GoalsPos.push_back(PlannerHNS::WayPoint(data.simuCars.at(i).x, data.simuCars.at(i).y, data.simuCars.at(i).z, data.simuCars.at(i).a));
+	}
+
+	return nData;
+}
+
+void GlobalPlanner::MainLoop()
+{
+	ros::Rate loop_rate(10);
+	timespec animation_timer;
+	UtilityHNS::UtilityH::GetTickCount(animation_timer);
+
+	while (ros::ok())
+	{
+		ros::spinOnce();
+		bool bMakeNewPlan = false;
+
+		if(m_params.mapSource == PlannerHNS::MAP_KML_FILE && !m_bKmlMap)
+		{
+			m_bKmlMap = true;
+			PlannerHNS::MappingHelpers::LoadKML(m_params.KmlMapPath, m_Map);
+			visualization_msgs::MarkerArray map_marker_array;
+			PlannerHNS::RosHelpers::ConvertFromRoadNetworkToAutowareVisualizeMapFormat(m_Map, map_marker_array);
+			pub_MapRviz.publish(map_marker_array);
+		}
+		else if (m_params.mapSource == PlannerHNS::MAP_FOLDER && !m_bKmlMap)
+		{
+			m_bKmlMap = true;
```

Comment:
I don't get this logic

Reply:
Load the specific map type once  in the begging of the node life time